### DISCUSSION TITLE
Schedule after Freeze Protect

### DISCUSSION THREAD

**Discussion by Tim-K5AC** [aqualinkd/AqualinkD#366] [General]
My pool just went into freeze protect this morning, when it got warm enough my AqualinkD schedule NEVER started.  Is this a bug, how do I insure that the schedule ALWAYS runs?

**Comment by sfeakes** [comment]
You might need to go into some more detail on exactly what was scheduled and times.  But I’m assuming that the freeze protect overrode the schedule.  Ie freeze protect kicked in at 6am off at 10am and schedule was turn pump on 9am off at 6pm.  The freeze protect turning the pump off at 10 am would do exactly that, there for pump would be off until next schedule to turn it on.

**Comment by Tim-K5AC** [comment]
Yes - Freeze Protect occurred during the event to turn on the pump.  I guess since AqualinkD is event driven (not state driven), the on command is ignored?  The only way to fix this is to add multiple on commands in the morning?

**Comment by sfeakes** [comment]
AqualinkD has a similar problem during a power outage, ie if the Control Panel restarts due to power failure when the pump is on, it will startup and pump will stay off until next on cycle.

I did create a small startup script that can overcome this, but only if AqualinkD is rebooted at the same time the control panel is.  So it's a hack for the moment.

I've been thinking of ways to overcome this, but don't have a good solution yet. 

Playing with the control panel and using its scheduling, it seems that after FreezeProtect ends it will go back to normal schedule, but after something like SWG Boots ends it won't go back to normal schedule.

I'll see if I can put something in the next release for this.

**Comment by sfeakes** [comment]
I Just implemented a quick update.  It's in development but you can try if you want.  If you pull the latest dev release (3.5.1), and add the below to `aqualinkd.conf` it should work.  Hopefully self explanatory.   

```
# Check if button_01 (usually Pump) is scheduled to run after an event that may have turned it off, and set it to run.
# Only for RS panels, Will not work for PDA panles.
# Example below is if pump is off due to power reset, freezeprotect or swg boots is turned off between 6am and 11pm then turn the pump on.
scheduler_check_pumpon_hour = 6
scheduler_check_pumpoff_hour = 23
scheduler_check_poweron = yes
scheduler_check_freezeprotectoff = yes
scheduler_check_boostoff = yes
```

I'll look to tie in on hour and off hour to the actual scheduler in the future.
BTW the test for above would be if current hour is greater than or equal 6 and less than 23.   So times of 6:00 and 22:59 pump would turn on, but 23:00 it would not.

**Reply by Tim-K5AC** [reply]
Thanks!  I will try it out this weekend.  We don't have any cold weather projected until next Tuesday/Wednesday.

**Comment by Tim-K5AC** [comment]
I was finally able to test this version after a freeze early Sunday morning, when we came back in the afternoon (temp's in the 40's) the pump was running (this would not have happened with the previous version).  The only strange thing I noticed, was the pump RPM was still running at the Freeze Protect RPM (1750) instead of the programmed value of 2500.  I don't suppose there is a way to make it run at the scheduled RPM?

**Reply by sfeakes** [reply]
Glad it worked.  Let me think about the RPM. All AqualinkD does it turn the pump on, it’s the control panel that will set the speed. So it will default to the base RPM speed set at the panel.
